apps/jobbrowser/src/jobbrowser/api2.py

```diff
@@ -216,16 +188,58 @@ def query_store_api(request, path=None):
 
 @api_error_handler
 def query_store_download_bundle(request, id=None):
-  response = {}
+  client = _create_query_store_client(request, content_type='application/octet-stream')
 
-  client = HttpClient(QUERY_STORE.SERVER_URL.get())
+  try:
+    zip_resp = client.execute("GET", 'api/data-bundle/' + id, stream=True);
+    response = StreamingHttpResponse(_zip_generate(zip_resp), content_type=zip_resp.headers.get('Content-Type'))
+    response['Content-Disposition'] = zip_resp.headers.get('Content-Disposition')
+    return response
+  except RestException as e:
+    return _handle_query_store_exp(e)
+
+
+def _query_store_proxy(request, path=None):
+  client = _create_query_store_client(request)
   resource = Resource(client)
+
+  try:
+    return resource.invoke(request.method, path, request.GET.dict(), request.body)
+  except RestException as e:
+    return _handle_query_store_exp(e)
+
+def _create_query_store_client(request, content_type='application/json; charset=UTF-8'):
+  headers = {
+    'x-do-as': request.user.username,
+    'X-Requested-By': 'das',
+    'Content-Type': content_type
+  }
+
+  client = HttpClient(QUERY_STORE.SERVER_URL.get())
+  client.set_headers(headers)
+
   if USE_SASL.get():
     client.set_kerberos_auth()
 
-  app = resource.get('api/data-bundle/' + id)
+  return client
 
-  response = FileResponse((app, 'rb'), content_type='application/octet-stream')
-  response['Content-Disposition'] = 'attachment; filename=' + id + '.zip'
+def _handle_query_store_exp(e):
```

Comment:
`exp` means?

Reply:
Renamed to _handle_query_store_exception